src/grpc/qgrpcchannel.cpp

```diff
@@ -0,0 +1,343 @@
+    );
+
+    *connection = QObject::connect(call.get(), &QGrpcChannelCall::finished, reply, [call, reply, connection, abortConnection](){
+        if (call->status.code() == QGrpcStatus::Ok) {
+            reply->setData(call->response);
+            reply->finished();
+        } else {
+            reply->setData({});
+            reply->error(call->status);
+        }
+
+        QObject::disconnect(*connection);
+        QObject::disconnect(*abortConnection);
+    });
+
+    *abortConnection = QObject::connect(reply, &QGrpcAsyncReply::error, call.get(), [call, connection, abortConnection](const QGrpcStatus &status){
+        if (status.code() == QGrpcStatus::Aborted) {
+            QObject::disconnect(*connection);
+            QObject::disconnect(*abortConnection);
+        }
+    });
+
+    call->start();
+}
+
+QGrpcStatus QGrpcChannelPrivate::call(const QString &method, const QString &service, const QByteArray &args, QByteArray &ret) {
+    QEventLoop loop;
+
+    QString rpcName = QString("/%1/%2").arg(service).arg(method);
+    QGrpcChannelCall call(m_channel.get(), rpcName, args);
+
+    QObject::connect(&call, &QGrpcChannelCall::finished, &loop, &QEventLoop::quit);
+
+    call.start();
+    loop.exec();
+
+    ret = call.response;
+    return call.status;
+}
+
+void QGrpcChannelPrivate::subscribe(QGrpcSubscription *subscription, const QString &service, QAbstractGrpcClient *client)  {
+    assert(subscription != nullptr);
+
+    QString rpcName = QString("/%1/%2").arg(service).arg(subscription->method());
+
+    std::shared_ptr<QGrpcChannelSubscription> sub;
+    std::shared_ptr<QMetaObject::Connection> abortConnection(new QMetaObject::Connection);
+    std::shared_ptr<QMetaObject::Connection> readConnection(new QMetaObject::Connection);
+    std::shared_ptr<QMetaObject::Connection> clientConnection(new QMetaObject::Connection);
+    std::shared_ptr<QMetaObject::Connection> connection(new QMetaObject::Connection);
+
+    sub.reset(
+        new QGrpcChannelSubscription(m_channel.get(), rpcName, subscription->arg(), subscription),
+        [](QGrpcChannelSubscription * sub) { sub->deleteLater(); }
+    );
+
+    *readConnection = QObject::connect(sub.get(), &QGrpcChannelSubscription::dataReady, subscription, [subscription](const QByteArray &data) {
+        subscription->handler(data);
+    });
+
+    *connection = QObject::connect(sub.get(), &QGrpcChannelSubscription::finished, subscription, [sub, subscription, readConnection, abortConnection, service, connection, clientConnection](){
+        qProtoDebug() << "Subscription ended with server closing connection";
+
+        QObject::disconnect(*connection);
+        QObject::disconnect(*readConnection);
+        QObject::disconnect(*abortConnection);
+        QObject::disconnect(*clientConnection);
+
+        switch (sub->status.code()) {
+            case QGrpcStatus::Ok:
+                break;
+            default:
+                qDebug()<<sub->status.code();
```

Comment:
Please use qProtoDebug instead. Add spaces arround <<

Reply:
That was actually supposed not to be there...